manager/src/main/java/org/openremote/manager/alarm/AlarmService.java

```diff
@@ -0,0 +1,504 @@
+        }
+
+        if (!identityProvider.isMasterRealmAdmin(userId) &&
+                (realms.size() > 1 || !identityProvider.isUserInRealm(userId, realms.stream().findFirst().orElse(null)))) {
+            throw new SecurityException("Realm is not active or inaccessible");
+        }
+    }
+
+    /**
+     * Sends an alarm if the given user has access to the alarm realm.
+     */
+    public SentAlarm sendAlarm(Alarm alarm, String userId) {
+        Objects.requireNonNull(alarm, "Alarm cannot be null");
+        Objects.requireNonNull(alarm.getRealm(), "Alarm realm cannot be null");
+        Objects.requireNonNull(alarm.getTitle(), "Alarm title cannot be null");
+        Objects.requireNonNull(alarm.getSeverity(), "Alarm severity cannot be null");
+        Objects.requireNonNull(alarm.getSource(), "Source cannot be null");
+        Objects.requireNonNull(alarm.getSourceId(), "Source ID cannot be null");
+
+        validateRealmAccessibleToUser(userId, alarm.getRealm());
+
+        long timestamp = timerService.getCurrentTimeMillis();
+
+        SentAlarm result = persistenceService.doReturningTransaction(entityManager -> {
+            SentAlarm sentAlarm = new SentAlarm()
+                    .setAssigneeId(alarm.getAssigneeId())
+                    .setRealm(alarm.getRealm())
+                    .setTitle(alarm.getTitle())
+                    .setContent(alarm.getContent())
+                    .setSeverity(alarm.getSeverity())
+                    .setStatus(alarm.getStatus())
+                    .setSource(alarm.getSource())
+                    .setSourceId(alarm.getSourceId())
+                    .setCreatedOn(new Date(timestamp))
+                    .setLastModified(new Date(timestamp));
+
+            entityManager.merge(sentAlarm);
+
+            return sentAlarm;
+        });
+
+        if (result != null) {
+            clientEventService.publishEvent(new AlarmEvent(alarm.getRealm(), PersistenceEvent.Cause.CREATE));
+            if (alarm.getSeverity() == Alarm.Severity.HIGH) {
+                Set<String> excludeUserIds = alarm.getSource() == MANUAL ? Set.of(alarm.getSourceId()) : Set.of();
+                sendAssigneeNotification(result, excludeUserIds);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Sends e-mail and push notifications for an alarm to the alarm assignee. If an assignee is not set all users having
+     * the {@link Constants#READ_ADMIN_ROLE} or{@link Constants#WRITE_ALARMS_ROLE} are notified.
+     *
+     * @param alarm the alarm to send e-mail and push notifications for
+     * @param excludeUserIds users matching these user IDs will are excluded from the notifications
+     */
+    protected void sendAssigneeNotification(SentAlarm alarm, Set<String> excludeUserIds) {
+        ManagerIdentityProvider identityProvider = identityService.getIdentityProvider();
+
+        List<User> users = new ArrayList<>();
+        if (alarm.getAssigneeId() == null) {
```

Comment:
Looks like `clientRoles` will need to be added to `UserQuery` if this behaviour is required